Source/JavaScriptCore/builtins/JSIteratorPrototype.js

```diff
@@ -212,12 +233,14 @@ function some(predicate)
     if (!@isObject(this))
         @throwTypeError("Iterator.prototype.some requires that |this| be an Object.");
 
-    if (!@isCallable(predicate))
+    var iterated = this;
```

Comment:
Let's just use `this` directly in this function instead of `iterated` when possible.